src/index.js

```diff
@@ -40,33 +61,90 @@ module.exports = class Client {
     this[PRIVATE.OFFSETS] = new Map()
     this[PRIVATE.LOGGER] = createLogger({ level: logLevel, logCreator })
     this[PRIVATE.CLUSTER_RETRY] = retry
-    this[PRIVATE.CREATE_CLUSTER] = ({
-      metadataMaxAge,
-      allowAutoTopicCreation = true,
-      maxInFlightRequests = null,
-      instrumentationEmitter = null,
-      isolationLevel,
-    }) =>
-      new Cluster({
+    this[PRIVATE.CREATE_BROKERPOOL] = (
+      {
+        metadataMaxAge = DEFAULT_METADATA_MAX_AGE,
+        allowAutoTopicCreation = true,
+        maxInFlightRequests = null,
+        instrumentationEmitter = null,
+      },
+      BrokerPoolType = BrokerPool
+    ) => {
+      const connectionBuilder = createConnectionBuilder({
         logger: this[PRIVATE.LOGGER],
         retry: this[PRIVATE.CLUSTER_RETRY],
-        offsets: this[PRIVATE.OFFSETS],
+        instrumentationEmitter,
         socketFactory,
         brokers,
         ssl,
         sasl,
         clientId,
         connectionTimeout,
-        authenticationTimeout,
-        reauthenticationThreshold,
         requestTimeout,
         enforceRequestTimeout,
+        maxInFlightRequests,
+      })
+      return new BrokerPoolType({
```

Comment:
Do we really need to have two different implementations here? What I'm thinking is, is there any difference between a connection pool that is not shared with any other instance, and just having a regular brokerpool? From what I can see, the only difference is in instrumentation event dispatching, but is there actually a benefit to not using that same code for "non-shared" connectionpools?